packages/core/src/components/tabs/tabs.tsx

```diff
@@ -141,15 +157,72 @@ export class Tabs extends AbstractPureComponent<ITabsProps, ITabsState> {
                     role="tablist"
                 >
                     {tabIndicator}
-                    {tabTitles}
+                    {this.props.overflow ? (
+                        <OverflowList
+                            items={tabTitles}
+                            visibleItemRenderer={this.visibleItemsRenderer}
+                            ref={this.refHandlers.overflowList}
+                            {...this.props.overflowListProps}
+                            overflowRenderer={this.overflowRenderer}
+                        />
+                    ) : (
+                        tabTitles
+                    )}
                 </div>
                 {tabPanels}
             </div>
         );
     }
 
+    public overflowRenderer = (items: Array<React.ReactElement<ITabProps>>) => {
+        const hasOverflowRenderer = this.props.overflowListProps && this.props.overflowListProps.overflowRenderer;
+
+        const overflowMenu = (
+            <Menu>
+                {items.map(item => {
+                    if (item.props.id) {
+                        const onClick = (e: React.MouseEvent<HTMLElement>) => {
+                            item.props.onClick(item.props.id, e);
+                        };
+
+                        return (
+                            <MenuItem
+                                key={item.props.id}
+                                text={item.props.title}
+                                onClick={onClick}
+                                disabled={item.props.disabled}
+                                active={item.props.selected}
+                            />
+                        );
+                    } else {
+                        return item;
+                    }
+                })}
+            </Menu>
+        );
+
+        return (
+            <div className={OVERFLOW_CONTENT_CLASS}>
+                {hasOverflowRenderer ? (
+                    this.props.overflowListProps.overflowRenderer
+                ) : (
+                    <Popover content={overflowMenu}>
+                        <span style={{ lineHeight: "30px", cursor: "pointer" }}>Overflow</span>
+                    </Popover>
+                )}
+            </div>
+        );
+    };
+
+    public visibleItemsRenderer = (item: React.ReactChild) => {
+        return item;
+    };
+
     public componentDidMount() {
-        this.moveSelectionIndicator();
+        // requires overflowList to render in order for it to figure out what elements are overflowing
+        window.requestAnimationFrame(() => {
+            this.moveSelectionIndicator();
+        });
```

Comment:
please move the `rAF` call _inside_ `this.moveSelectionIndicator()` because you do it every time now.